app/code/Magento/ConfigurableProductGraphQl/Model/Options/Collection.php

```diff
@@ -49,16 +72,28 @@ class Collection
     /**
      * @param CollectionFactory $attributeCollectionFactory
      * @param ProductFactory $productFactory
+     * @param ProductRepository $productRepository
      * @param MetadataPool $metadataPool
+     * @param Data $configurableProductHelper
+     * @param Metadata $optionsMetadata
+     * @param SelectionUidFormatter $selectionUidFormatter
      */
     public function __construct(
         CollectionFactory $attributeCollectionFactory,
         ProductFactory $productFactory,
-        MetadataPool $metadataPool
+        ProductRepository $productRepository,
+        MetadataPool $metadataPool,
+        Data $configurableProductHelper,
+        Metadata $optionsMetadata,
+        SelectionUidFormatter $selectionUidFormatter
     ) {
         $this->attributeCollectionFactory = $attributeCollectionFactory;
         $this->productFactory = $productFactory;
+        $this->productRepository = $productRepository;
         $this->metadataPool = $metadataPool;
+        $this->configurableProductHelper = $configurableProductHelper;
+        $this->optionsMetadata = $optionsMetadata;
```

Comment:
Please check constructor params for null to add new classes in a backward compatible way.

Reply:
Could you give me an example how it will look like?

Reply:
sure. 
`$this->searchApiCriteriaBuilder = $searchApiCriteriaBuilder ??
            \Magento\Framework\App\ObjectManager::getInstance()->get(SearchCriteriaBuilder::class);`

Reply:
added.